src/scalar.rs

```diff
@@ -48,6 +50,25 @@ use util::CTAssignable;
 #[derive(Copy,Clone)]
 pub struct Scalar(pub [u8; 32]);
 
+// XXX make CTEq traits
+impl Eq for Scalar{}
+impl PartialEq for Scalar {
+    /// Test equality between two `Scalar`s in constant time.
```

Comment:
I don't think this should be advertised as being "in constant time", since it returns a bool...